lib/console_web/controllers/router/device_controller.ex

```diff
@@ -36,23 +36,34 @@ defmodule ConsoleWeb.Router.DeviceController do
   end
 
   def show(conn, %{"id" => id}) do
-    case Devices.get_device(id) |> Repo.preload([:multi_buy, labels: [:multi_buy]]) do
+    case Devices.get_device(id) |> Repo.preload([:multi_buy, :config_profile, labels: [:multi_buy]]) do
       %Device{} = device ->
-        # TODO replace adr_allowed and cf_list_enabled values with those coming from config profile
-        adr_allowed =
+        config_profile =
           case length(device.labels) do
-            0 -> device.adr_allowed
+            0 ->
+              if device.config_profile do device.config_profile else nil end
             _ ->
-              label_has_adr_allowed = device.labels |> Enum.map(fn l -> l.adr_allowed end) |> Enum.any?(fn s -> s == true end)
-              if device.adr_allowed == true or label_has_adr_allowed do true else false end
+              device_labels =
+                Labels.get_labels_of_device_in_order_of_attachment(device)
+                |> Repo.preload([label: [:config_profile]])
+              first_device_label_with_config_profile = Enum.find(device_labels,fn dl -> dl.label.config_profile_id != nil end)
+              if first_device_label_with_config_profile do
+                first_device_label_with_config_profile.label.config_profile
+              else
+                if device.config_profile do [device.config_profile] else nil end
```

Comment:
should this be [device.config_profile]? not just device.config_profile

Reply:
oops fixed it in the other one but forgot this one, thank you!